### PR TITLE
[10.0][FIX] sale_operating_unit

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Basic sales user is not able to create SO because the code is searching on sales_team

It solves this issue https://github.com/OCA/operating-unit/issues/93

### CODE REVIEW

File: sale_operating_unit/models/sale.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class SaleOrder(models.Model):
 
     @api.model
     def _default_operating_unit(self):
-        team = self.env['crm.team']._get_default_team_id()
+        team = self.env['crm.team'].sudo()._get_default_team_id()
```
Comment (MiquelRForgeFlow) [inline sale_operating_unit/models/sale.py]: This sudo makes absolutely sense to me. The rest, no.

Comment (AaronHForgeFlow) [reply sale_operating_unit/models/sale.py]: what about self.team_id.sudo() ?
If we remove the sudos completely users will get an AccessError

Comment (MiquelRForgeFlow) [reply sale_operating_unit/models/sale.py]: why? the sudo is for accessing methods, not values. sudo is for browse, search, etc.

Comment (AaronHForgeFlow) [reply sale_operating_unit/models/sale.py]: I don't know :see_no_evil:  

The thing is I got AccessError if don't use that. I don't see any other solution.

Comment (MiquelRForgeFlow) [reply sale_operating_unit/models/sale.py]: Could you dive a bit deeper in the code while debugging to see the cause? :)

Comment (AaronHForgeFlow) [reply sale_operating_unit/models/sale.py]: The problem is the search in the sales team. Either we search using sudo, or we give permissions to the user.

Comment (MiquelRForgeFlow) [reply sale_operating_unit/models/sale.py]: I prefer the sudo in the search

**Comment by AaronHForgeFlow** [comment]
o-no-idea-dog-meme-facebook

**Comment by AaronHForgeFlow** [comment]
This issue is not actually happen or at least I cannot reproduce it. Closing

**Pull request closed without merging**